github/github.go

```diff
@@ -735,9 +735,12 @@ func category(path string) rateLimitCategory {
 // Deprecated: RateLimit is deprecated, use RateLimits instead.
 func (c *Client) RateLimit() (*Rate, *Response, error) {
 	limits, resp, err := c.RateLimits()
-	if limits == nil {
+	if err != nil {
 		return nil, nil, err
 	}
+	if limits == nil {
+		return nil, nil, fmt.Errorf("RateLimits returned nil limits and error. Unable to extract Core rate limit")
```

Comment:
About the error message, see https://github.com/golang/go/wiki/CodeReviewComments#error-strings.

I don't think you should separate it into multiple sentences and capitalize the letters. First letter happens to be capital because it refers to `RateLimits` the proper noun.